Wrap all amount parts when html option is used

rolandasb opened the pull request:
In response of #775 

When using formatter's `html: true` option, all amount's string parts
will be wrapped in html elements (symbol, whole part, decimal mark,
decimal part and, if `with_currency` is true, currency).

Also deprecated html_wrap_symbol option, since it will be always wrapped
with `html: true` option.

@antstorm I'm not sure about deprecation part though, because we potentially introduce breaking change here. Are you going to make a major release or should we introduce temporary option which enables this new type of behaviour?

antstorm commented on lib/money/money/formatter.rb:
```diff
@@ -132,11 +132,11 @@ class Formatter
     #   Money.new(100000, "FOO").format #=> "$1,000.00"
     #
     # @option rules [Boolean] :html (false) Whether the currency should be
```
Should say that every part should be HTML-formatted

antstorm commented:
@rolandasb great PR, left a couple on inline comments 👍

As for the deprecation — it might be worth adding a new option `html_wrap` while leaving existing options intact, but deprecating them. We don't want to break someone's UI with this.

rolandasb commented:
@antstorm A couple of updates:
1. Brought back old options and deprecated them.
2. Renamed new option to `html_wrap: true`.
3. Extracted HTML element wrapping to a separate private method to reduce duplication.

Let me know if there's anything else I can do 🙂
